windows.js

```diff
@@ -50,15 +54,70 @@ export async function install(platform, engine, version) {
 
   let toolchainPaths = (version === 'mswin') ? await setupMSWin() : await setupMingw(version)
 
-  common.setupPath([`${rubyPrefix}\\bin`, ...toolchainPaths])
-
   if (!inToolCache) {
     await downloadAndExtract(engine, version, url, base, rubyPrefix);
   }
 
+  const winMSYS2Type = common.setupPath([`${rubyPrefix}\\bin`, ...toolchainPaths])
+
+  // install msys2 tools for all Ruby versions, only install mingw or ucrt for Rubies >= 2.4
+
+  if (!['windows-2019', 'windows-2016'].includes(virtualEnv)) {
+    await installMSYS2Tools()
+  }
+
+  if ((( winMSYS2Type === 'ucrt64') || !['windows-2019', 'windows-2016'].includes(virtualEnv)) &&
+    (common.floatVersion(version) >= 2.4)) {
```

Comment:
Is that a condition for MSYS2 vs MSYS1? If so please extract in a variable which name indicates that (e.g., usesMSYS2)

Reply:
@MSP-Greg could you answer/fix? Also extra space in `( winMSYS2Type` above

Reply:
Sorry, fixed the space and added a comment...